tools/isopack.js

```diff
@@ -919,6 +919,45 @@ _.extend(Isopack.prototype, {
     });
   },
 
+  _canWriteLegacyBuilds(options) {
+    var writeLegacyBuilds = false;
+
+    if (options.includePreCompilerPluginIsopackVersions) {
+      // We will reset this to false if at any point later we determine that
+      // this package cannot be saved in the legacy format (because it uses a
+      // compiler plugin other than JS or CSS).
+      writeLegacyBuilds = true;
+    }
+
+    function isResourceUnsafeForLegacyBuilds(resource) {
+      if (resource.hasSourceProcessors) {
+        // If this resource is processed by a new-style compiler plugin,
+        // then it cannot be written as part of a legacy Isopack.
+        return true;
+      }
+
+      if (resource.type === "source") {
+        // This package cannot be represented as an isopack-1 Isopack if
+        // it uses a file implemented by registerCompiler other than the
+        // very basic JS and CSS types.
+        return ! (
+          resource.extension === "js" ||
+          resource.extension === "css"
+        );
+      }
+
+      return false;
+    }
+
+    return writeLegacyBuilds && ! _.any(
+      this.unibuilds,
+      unibuild => _.any(
```

Comment:
think our style guide only lets you drop parens/curlies/return for single-line arrows?

also you could use Array.some if you want for both of these.